src/state/opamSwitchState.ml

```diff
@@ -1305,3 +1305,160 @@ let update_repositories gt update_fun switch =
   OpamFile.Switch_config.write
     (OpamPath.Switch.switch_config gt.root switch)
     conf
+
+
+(* dependencies computation *)
+
+let dependencies_filter_to_formula ~build ~post st nv formula =
+  let env v =
+    if List.mem v OpamPackageVar.predefined_depends_variables then
+      match OpamVariable.Full.to_string v with
+      | "build" -> Some (B build)
+      | "post" -> Some (B post)
+      | _ -> None
+    else
+      OpamPackageVar.resolve_switch ~package:nv st v
+  in
+  OpamFilter.filter_formula ~default:true env formula
+
+let dependencies ~depopts ~build ~post ~installed
+    ?(unavailable=false)
+    st universe packages =
+  if OpamPackage.Set.is_empty packages then OpamPackage.Set.empty else
+  let base =
+    packages ++
+    if installed then  universe.u_installed
+    else if unavailable then universe.u_packages
+    else universe.u_available
+  in
+  log ~level:3 "dependencies packages=%a"
+    (slog OpamPackage.Set.to_string) packages;
+  let timer = OpamConsole.timer () in
+  let base_depends =
+    let depends =
+      OpamPackage.Map.filter (fun nv _ -> OpamPackage.Set.mem nv base)
+        universe.u_depends
+    in
+    if depopts then
+      let depopts =
+        OpamPackage.Map.filter (fun nv _ -> OpamPackage.Set.mem nv base)
+          universe.u_depopts
+      in
+      OpamPackage.Map.union (fun d d' -> OpamFormula.And (d, d'))
+        depopts depends
+    else
+      depends
+  in
+  let open OpamPackage.Set.Op in
+  let get_deps nvs =
+    OpamPackage.Set.fold (fun nv acc ->
+        let depends_ff = OpamPackage.Map.find nv base_depends in
+        let depends_formula =
+          dependencies_filter_to_formula ~build ~post st nv depends_ff
+        in
+        if depends_formula = Empty then acc else
+        let packages = OpamFormula.packages base depends_formula in
+        if OpamPackage.Set.is_empty packages then acc else
+          packages ++ acc)
+      nvs OpamPackage.Set.empty
+  in
+  let rec aux all deps =
+    let new_deps = get_deps deps in
+    if OpamPackage.Set.is_empty new_deps then all
+    else aux (all ++ new_deps) (new_deps -- all)
+  in
+  let result = aux packages packages in
+  log "Dependencies (%.3f) result=%a" (timer ())
```

Comment:
Do we keep that at level 1? With #5208, it will be displayed at each switch state update

Reply:
Yes i think that's fine. It's debug anyway so if we find it's too much data we can always bump its level up later.